packages/core/src/App/Containers/RealAccountSignup/address-details.jsx

```diff
@@ -37,206 +37,183 @@ const InputField = props => {
     );
 };
 
-class AddressDetails extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            has_fetched_states_list: false,
-            address_state_to_display: '',
-        };
-        // TODO: Find a better solution for handling no-op instead of using is_mounted flags
-        this.is_mounted = false;
-    }
-
-    async componentDidMount() {
-        this.is_mounted = true;
-        await this.props.fetchStatesList();
-        if (this.is_mounted)
-            this.setState({
-                has_fetched_states_list: true,
-                address_state_to_display: getLocation(this.props.states_list, this.props.value.address_state, 'text'),
-            });
-    }
+const AddressDetails = ({
+    fetchStatesList,
+    states_list,
+    getCurrentStep,
+    onSave,
+    onCancel,
+    goToNextStep,
+    goToPreviousStep,
+    validate,
+    onSubmit,
+    is_svg,
+    is_gb_residence,
+    ...props
+}) => {
+    const [has_fetched_states_list, setHasFetchedStatesList] = React.useState(false);
+    const [address_state_to_display, setAddressStateToDisplay] = React.useState('');
 
-    componentWillUnmount() {
-        this.is_mounted = false;
-    }
+    React.useEffect(() => {
+        const fetchStateList = async () => {
+            await fetchStatesList();
+            setHasFetchedStatesList(true);
+            setAddressStateToDisplay(getLocation(states_list, props.value.address_state, 'text'));
+        };
+        fetchStateList();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
Is it possible to use `makeCancellablePromise` from `/packages/core/src/_common/base/cancellable_promise.js` and cancell it if the modal is closed before the state list is fetched?

Reply:
Sure, I put `fetchStateList` inside `makeCancellablePromise` and cancel it upon unmounting  of the component.